tests/src/Context/DrupalContext.php

```diff
@@ -50,4 +57,35 @@ public function visitContentPage($type, $title) {
     $this->visitPath($path);
   }
 
+  /**
+   * Remember the last node id.
+   *
+   * @BeforeScenario @reset-nodes
+   */
+  public function rememberCurrentLastNode() {
+    $query = db_select('node');
+    $query->addExpression('MAX(nid)');
+    $this->maxNodeId = $query->execute()->fetchField();
+  }
+
+  /**
+   * Removes any nodes created after the last node id remembered.
+   *
+   * @AfterScenario @reset-nodes
+   */
+  public function resetNodes() {
+    $max_nid = isset($this->maxNodeId) ? $this->maxNodeId : 0;
```

Comment:
By setting this to 0 if not set we will delete nodes in the system, right? The method name is a bit misleading then, since I would expect it to reset my site at the state before the scenario execution.

Anyway I would not delete all nodes but only those created in the scenario, the platform comes with node 1 which is created at installation stage (the default editorial group) and this would delete that node, which is annoying for developers mainly.